xbundle/__init__.py

```diff
@@ -1,596 +1,512 @@
+        """
         self.course = etree.Element('course')
         self.metadata = etree.Element('metadata')
         self.urlnames = []
-        self.xml = None				# only used if XML xbundle file was read in
+        self.xml = None  # only used if XML xbundle file was read in
         self.keep_urls = keep_urls
-        self.force_studio_format = force_studio_format	# sequential must be followed by vertical in export
+        # sequential must be followed by vertical in export
+        self.force_studio_format = force_studio_format
         self.skip_hidden = skip_hidden
         self.keep_studio_urls = keep_studio_urls
         self.no_overwrite = no_overwrite or []
-        self.overwrite_files = []
-        return
-
-        
-    #----------------------------------------
-    # creation by parts
+        self.path = ""
+        self.semester = ""
+        self.export = None
 
-    def set_course(self,xml):
-        if not xml.tag=='course':
-            self.errlog('set_course should be called with a <course> element')
+    def set_course(self, xml):
+        """
+        Set self.course from the XML passed in.
+        """
+        if not xml.tag == 'course':
+            log.error('set_course should be called with a <course> element')
             return
         if not 'org' in xml.attrib:
-            xml.set('org',self.DefaultOrg)
-        semester = xml.get('url_name', xml.get('semester', self.DefaultSemester))
+            xml.set('org', "MITx")
+        semester = xml.get('url_name', xml.get('semester', ''))
+        if semester == "":
+            log.error("No semester found.")
         if not 'semester' in xml.attrib:
             xml.set('semester', semester)
         self.semester = semester
         self.course = xml
-        # fill up self.urlnames with existing ones if keep_urls
+        # Fill up self.urlnames with existing ones if keep_urls.
         if self.keep_urls:
             def walk(xml):
-                un = xml.get('url_name','')
-                if un:
-                    self.urlnames.append(un)
-                if xml.tag in self.DescriptorTags:
+                """
+                Recursively walk XML and gather url names.
+                """
+                url_name = xml.get('url_name', '')
+                if url_name:
+                    self.urlnames.append(url_name)
+                if xml.tag in DESCRIPTOR_TAGS:
                     for child in xml:
                         walk(child)
             walk(xml)
-    
-        
+
     def add_policies(self, policies):
-        '''add a policies XML subtree to the metadata'''
+        """add a policies XML subtree to the metadata"""
         self.metadata.append(policies)
 
-    
-    def set_about(self, about):
-        '''set about XML tree'''
-        xabout = self.metadata.find('about')
-        if xbout is not None:
-            self.metadata.remove(xabout)
-        self.metadata.append(xabout)
-
-
```

Comment:
Yes, also not used. And nobody else was using it, guaranteed, because it contained a typo (`about` instead of `xabout`).